REFRAME.md

```diff
@@ -36,24 +36,67 @@ Given that a client C wants to request information from some server S:
 1. C opens sends a request to S that is a single message
 2. S will respond to C with either a single message or a group of messages
 
+## Cachability
+
+Some methods are prone to being cachable while others are not. Methods are tagged within the spec as cachable/not cachable. Transports may leverage that information to decide if cachable/not cachable methods should be treated differently.
+
 ## Transports
 
 ### HTTP + DAG-JSON
 
-All messages MUST be encoded as DAG-JSON and use explicit content type `application/vnd.ipfs.rpc+dag-json; version=1`
+All messages sent in HTTP body MUST be encoded as DAG-JSON and use explicit content type `application/vnd.ipfs.rpc+dag-json; version=1`
 
+Requests MUST be sent as either:
+- `GET /reframe/{mbase64url-dag-cbor}`
+  - Cacheable HTTP `GET` requests with message passed as DAG-CBOR in HTTP path segment, encoded as URL-safe [`base64url` multibase](https://docs.ipfs.io/concepts/glossary/#base64url) string
+  - Suitable for sharing links, sending smaller messages, and when a query result MUST benefit from HTTP caching (see _HTTP Caching Considerations_ below).
+- `POST /reframe`
+  - Ephemeral HTTP `POST` request with message passed as DAG-JSON in HTTP request body
+  - Suitable for bigger messages, and when HTTP caching should be skipped for the most fresh results
 
-Requests MUST be sent as HTTP POST requests. If a server supports HTTP/1.1, then it MAY send chunked-encoded messages. Clients supporting HTTP/1.1 MUST accept chunked-encoded responses.
+Severs MUST support `GET` for methods marked as cacheable and MUST support `POST` for methods marked as non-cacheable.

```

Comment:
@lidel I added this to deal with https://github.com/ipfs/specs/pull/287/files#r891805528 and because I think this is what we talked about https://github.com/ipfs/specs/pull/287/files#r899107207.

lmk if you disagree

Reply:
I think this is fine since the MUST requirement for cacheable GETs means clients won't have to guess which to use, but they'll have a fallback.

@willscott any objections to also supporting POST here?

Reply:
should the server be able to choose to respond with 'method not supported' to posts when it is under heavy load and needs to offload to a cache. 

are we allowed to mitigate clients that don't attempt a GET and go directly to a POST?

Reply:
@willscott  Yes, HTTP service can decide to refuse.. service ;-) but there are better ways.

HTTP servers can return  [429 Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429) and [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) to limit use of expensive POST while encouraging GET for cachable Reframe methods.

Reply:
I've added a section about rate-limiting [here](https://github.com/ipfs/specs/pull/287/commits/62eb11e7e93c5972665c4bf3c38cbedac41e9ea8?short_path=08fb031#diff-08fb031656230573a5a243374d2f35a25fc604ca11aa36b96c27b1f96aa3f9be)